torchgeo/samplers/single.py

```diff
@@ -185,3 +186,11 @@ def __iter__(self) -> Iterator[BoundingBox]:
                     maxx = minx + self.size[1]
 
                     yield BoundingBox(minx, maxx, miny, maxy, mint, maxt)
+
+    def __len__(self) -> int:
```

Comment:
What's the use case for having a `__len__` method?

Reply:
I don't think I'm understanding your question correctly. The obvious use case is when a user wants to know how many patches will be sampled in the grid sampler... You can use this information to show progress bars, benchmark time per patch without having to keep a running count of the patches, to satisfy your curiosity, etc.

Reply:
Gotcha, I've never tried to add a progress bar before but it's not a bad idea. I think the reason I didn't add a `__len__` method in the first place was that I was worried about efficiency, but I think it's fine.